src/script_opt/CPP/RuntimeInits.cc

```diff
@@ -383,11 +383,16 @@ TypePtr CPP_TypeInits::BuildRecordType(InitsManager* im, ValElemVec& init_vals,
     auto r = cast_intrusive<RecordType>(inits_vec[offset]);
     ASSERT(r);
 
-    if ( r->NumFields() == 0 ) {
+    auto addl_fields = init_vals[2];
+
+    if ( addl_fields > 0 || r->NumFields() == 0 ) {
+        // We shouldn't be adding fields if the record doesn't have any
+        // existing fields - that would reflect an initialization botch.
+        ASSERT(addl_fields == 0 || r->NumFields() > 0);
```

Comment:
`FatalError()` or `InternalError`? This shouldn't be hot, so better to fail in non-debug bulids, too, no?

Reply:
Good point, fixed in the fixup I just pushed